### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by vmatyusGitHub**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Add migration guide from `nodejs-cloudant` to new beta `cloudant-node-sdk` library.
Guide includes a reference mapping table that links nodejs-cloudant to cloudant-node-sdk API operation documentation link.

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Migrating to the `cloudant-node-sdk` library
+This document is to assist in migrating from the `nodejs-cloudant` (module: `cloudant`) to the newly supported [`cloudant-node-sdk`](https://github.com/IBM/cloudant-node-sdk) (package: `@ibm-cloud/cloudant`). Compatible with Javascript and TypeScript.
```
Comment (bessbd) [inline MIGRATION.md:2]: Nit:
```suggestion
This document is to assist in migrating from the `nodejs-cloudant` (module: `cloudant`) to the newly supported [`cloudant-node-sdk`](https://github.com/IBM/cloudant-node-sdk) (package: `@ibm-cloud/cloudant`) that is supported in Javascript and TypeScript.
```

File: README.md [hunk 2]
```diff
@@ -1309,6 +1310,10 @@ suite, just run `npm test`.
 
 You can also run the tests with verbose output using `npm test-verbose`
 
+## Migrating to `cloudant-node-sdk` library
+We have a newly supported Cloudant Python SDK named [cloudant-node-sdk](https://github.com/IBM/cloudant-node-sdk).
```
Comment (bessbd) [inline README.md:1314]:
```suggestion
We have a newly supported Cloudant Node.js SDK named [cloudant-node-sdk](https://github.com/IBM/cloudant-node-sdk).
```

File: MIGRATION.md (hunk 1 above)
Comment (emlaver) [inline MIGRATION.md:2]: Let's use package for both libraries, and the legacy package is named `@cloudant/cloudant`
```suggestion
This document is to assist in migrating from the `nodejs-cloudant` (package: `@cloudant/cloudant`) to the newly supported [`cloudant-node-sdk`](https://github.com/IBM/cloudant-node-sdk) (package: `@ibm-cloud/cloudant`). Compatible with Javascript and TypeScript.
```

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-node-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-node-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-node-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-node-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. For storing credentials in a file, `dotenv` package not needed anymore. Check the new operation [here](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration).
```
Comment (emlaver) [inline MIGRATION.md:13]: How about:
```suggestion
1. For storing credentials in a file, the `dotenv` package is not supported. See the [external file configuration section](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration) in our API docs for handling this feature in our new library.
```

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+1. No need to fetch database object in advance to perform operations on Documents. For example, in the case of the document update, this would mean calling first `getDocument` to fetch and after `putDocument` to update.
```
Comment (emlaver) [inline MIGRATION.md:14]: Does this apply to just documents?
```suggestion
1. Fetching the database object first before performing operations on documents is not required. For example, in the case of updating a document, this would mean calling first `getDocument` to fetch and after `putDocument` to update.
```

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+1. There are no plugin handling anymore.
```
Comment (emlaver) [inline MIGRATION.md:15]: We might want to say that plugin handling isn't supported but several of the plugins exist in the new library.  We might also want to give an example of a plugin (e.g. IAM) that is supported e.g.
```suggestion
1. Plugins are not supported, but several of the plugin features exist in the new library e.g. IAM.
```

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+1. Authentication mode can be set [this way](https://github.com/IBM/cloudant-node-sdk/tree/master#authentication).
```
Comment (emlaver) [inline MIGRATION.md:16]: I think we can remove this since we already have a section about for authentication.

File: MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+1. Retries currently not supported.
```
Comment (emlaver) [inline MIGRATION.md:17]: Let's be a bit more specific about the retry feature:
```suggestion
1. The plugin for retrying failed requests is not supported.
```

**Review by emlaver — CHANGES_REQUESTED** [review]
Looking good

File: MIGRATION.md (hunk 1 above)
Comment (vmatyusGitHub) [reply MIGRATION.md:2]: This way `supported` word is used 2 times in a sentence. On the Hungarian Literature lesson taught me this is an invalid composition, better to use synonyms.
Okay, this is not a novel, but still. 
WDYT about this:
```suggestion
This document is to assist in migrating from the `nodejs-cloudant` (module: `cloudant`) to the newly supported [`cloudant-node-sdk`](https://github.com/IBM/cloudant-node-sdk) (package: `@ibm-cloud/cloudant`) that is applicable with Javascript and TypeScript.
```

File: README.md (hunk 2 above)
Comment (vmatyusGitHub) [reply README.md:1314]: Nice catch!
However, I would prefer to use just `Node`, because `ts` also supported.

Comment (bessbd) [reply README.md:1314]: image

I get that `js` might be slightly misleading there, but the SDK is called `IBM Cloudant Node.js SDK`. Same thing for the platform itself. It is called `node.js`  ( https://nodejs.org/en/ )

Not a big deal though. Just being pedantic.

File: MIGRATION.md (hunk 1 above)
Comment (bessbd) [reply MIGRATION.md:2]: Good call. I'm not an English teacher though :)

File: MIGRATION.md (hunk 3 above)
Comment (vmatyusGitHub) [reply MIGRATION.md:13]: For me `the `dotenv` package is not supported.` says that we deal with issues related to `dotenv` but this is not the case. This is just a suggested package in this situations. So I am reluctant about the first part. 

WDYT about this:
```suggestion
1. For storing credentials in a file has changed, the `dotenv` package is not needed anymore. See the [external file configuration section](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration) in our API docs for the new handling.
```

File: MIGRATION.md (hunk 4 above)
Comment (vmatyusGitHub) [reply MIGRATION.md:14]: Not really, there are a bunch of other things like: replicate, db info, view, find.
I also removed `on documents` from the first sentence.
```suggestion
1. Fetching the database object first before performing operations on it is not required. For example, in the case of updating a document, this would mean calling first `getDocument` to fetch and after `putDocument` to update.
```

File: MIGRATION.md (hunk 1 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Yeah, I checked this word, it is rather used in legal context. So switched back to `compatible` currently in c706430

File: README.md (hunk 2 above)
Comment (vmatyusGitHub) [reply README.md]: This was new for me. In this case I am okay with the suggestion. Changed in 8b5920f

File: MIGRATION.md (hunk 1 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in c706430

File: MIGRATION.md (hunk 3 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in c706430

File: MIGRATION.md (hunk 4 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in c706430

File: MIGRATION.md (hunk 5 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in c706430

File: MIGRATION.md (hunk 6 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Removed in c706430

File: MIGRATION.md (hunk 7 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in c706430

File: MIGRATION.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+There are several ways to create a client connection in `cloudant-node-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-node-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-node-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-node-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. The way of storing credentials in a file has changed, the usage of `dotenv` package is not suggested. See the [external file configuration section](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration) in our API docs about the new handling.
+1. Fetching the database object first before performing operations on it is not required. For example, in the case of updating a document, this would mean calling first `getDocument` to fetch and after `putDocument` to update.
```
Comment (emlaver) [inline MIGRATION.md:14]: `before performing operations on it` sounds a bit awkward.  What about:
```suggestion
1. Fetching the database object first before performing additional operations is not required. For example, in the case of updating a document you would first call `getDocument` to fetch and then `putDocument` to update.
```

File: MIGRATION.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+|`db.destroy()`|[deleteDatabase](https://cloud.ibm.com/apidocs/cloudant?code=node#deletedatabase)|
+|`db.info()`|[getDatabaseInformation](https://cloud.ibm.com/apidocs/cloudant?code=node#getdatabaseinformation)|
+|`db.insert()`|[postDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#postdocument)|
+|`db.create(db_name)`|[putDatabase](https://cloud.ibm.com/apidocs/cloudant?code=node#putdatabase)|
+|`db.fetch()/db.list()/db.listAsStream()`|[postAllDocs](https://cloud.ibm.com/apidocs/cloudant?code=node#postalldocs)|
+|`db.bulk()`|[postBulkDocs](https://cloud.ibm.com/apidocs/cloudant?code=node#postbulkdocs)|
+|`db.bulk_get()`|[postBulkGet](https://cloud.ibm.com/apidocs/cloudant?code=node#postbulkget)|
+|`db.changes()`|[postChanges](https://cloud.ibm.com/apidocs/cloudant?code=node#postchanges)|
+|`db.destroy()`|[deleteDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#deletedesigndocument)|
+|`db.get()`|[getDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#getdesigndocument)|
```
Comment (emlaver) [inline MIGRATION.md:59]:
```suggestion
|`db.get() with _design path`|[getDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#getdesigndocument)|
```

File: MIGRATION.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+|`db.insert()`|[putDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#putdesigndocument)|
```
Comment (emlaver) [inline MIGRATION.md:60]:
```suggestion
|`db.insert() with _design path`|[putDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#putdesigndocument)|
```

File: MIGRATION.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+|`uuids()`|[getUuids](https://cloud.ibm.com/apidocs/cloudant?code=node#getuuids)|
+|`db.destroy()`|[deleteDatabase](https://cloud.ibm.com/apidocs/cloudant?code=node#deletedatabase)|
+|`db.info()`|[getDatabaseInformation](https://cloud.ibm.com/apidocs/cloudant?code=node#getdatabaseinformation)|
+|`db.insert()`|[postDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#postdocument)|
+|`db.create(db_name)`|[putDatabase](https://cloud.ibm.com/apidocs/cloudant?code=node#putdatabase)|
+|`db.fetch()/db.list()/db.listAsStream()`|[postAllDocs](https://cloud.ibm.com/apidocs/cloudant?code=node#postalldocs)|
+|`db.bulk()`|[postBulkDocs](https://cloud.ibm.com/apidocs/cloudant?code=node#postbulkdocs)|
+|`db.bulk_get()`|[postBulkGet](https://cloud.ibm.com/apidocs/cloudant?code=node#postbulkget)|
+|`db.changes()`|[postChanges](https://cloud.ibm.com/apidocs/cloudant?code=node#postchanges)|
+|`db.destroy()`|[deleteDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#deletedesigndocument)|
```
Comment (emlaver) [inline MIGRATION.md:58]: I think we'll want to include the _design path for these operations:
```suggestion
|`db.destroy() with _design path`|[deleteDesignDocument](https://cloud.ibm.com/apidocs/cloudant?code=node#deletedesigndocument)|
```

**Review by emlaver — COMMENTED** [review]
Looks good, some final comments

File: MIGRATION.md (hunk 1 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Today I found an other word: `compliant` WDYT?

File: MIGRATION.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-node-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-node-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-node-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-node-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. The way of storing credentials in a file has changed, the usage of `dotenv` package is not suggested. See the [external file configuration section](https://github.com/IBM/cloudant-node-sdk#authentication-with-external-configuration) in our API docs about the new handling.
```
[the remark this answers is not in the document]
Comment (vmatyusGitHub) [reply MIGRATION.md]: Yes, this looks good now. Changed: 583c25f

File: MIGRATION.md (hunk 8 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Okay. Changed 583c25f

File: MIGRATION.md (hunk 9 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in 583c25f

File: MIGRATION.md (hunk 10 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in 583c25f

File: MIGRATION.md (hunk 11 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in 583c25f

File: MIGRATION.md (hunk 1 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Check it here: 583c25f

File: MIGRATION.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
+# Migrating to the `cloudant-node-sdk` library
+This document is to assist in migrating from the `nodejs-cloudant` (package: `@cloudant/cloudant`) to the newly supported [`cloudant-node-sdk`](https://github.com/IBM/cloudant-node-sdk) (package: `@ibm-cloud/cloudant`) that is JavaScript and TypeScript compliant.
```
Comment (emlaver) [inline MIGRATION.md:2]: Out of curiosity, why did you change compatible with compliant? Compatible seems to be the better word for the relationship between `cloudant-node-sdk` and JS/TS.

**Review by emlaver — APPROVED** [review]
Looks great!

File: MIGRATION.md (hunk 13 above)
Comment (vmatyusGitHub) [reply MIGRATION.md:2]: Okay, then I revert it back :)

File: MIGRATION.md (hunk 1 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Reverted this back: 40dd63a

File: MIGRATION.md (hunk 13 above)
Comment (vmatyusGitHub) [reply MIGRATION.md]: Changed in 40dd63a

**Comment by vmatyusGitHub** [comment]
Thank you for reviewing @emlaver and @bessbd ! I am about to merge this.
